Index: Makefile
===================================================================
RCS file: /cvs/ports/devel/quirks/Makefile,v
retrieving revision 1.44
diff -u -r1.44 Makefile
--- Makefile	6 May 2011 11:28:55 -0000	1.44
+++ Makefile	13 May 2011 00:10:54 -0000
@@ -5,7 +5,7 @@
 DISTFILES =
 
 # API.rev
-PKGNAME = 	quirks-1.42
+PKGNAME = 	quirks-1.43
 PKG_ARCH =	*
 MAINTAINER =	Marc Espie <espie@openbsd.org>
 
Index: files/Quirks.pm
===================================================================
RCS file: /cvs/ports/devel/quirks/files/Quirks.pm,v
retrieving revision 1.50
diff -u -r1.50 Quirks.pm
--- files/Quirks.pm	6 May 2011 11:28:55 -0000	1.50
+++ files/Quirks.pm	13 May 2011 00:10:54 -0000
@@ -226,6 +226,8 @@
 sub is_base_system
 {
 	my ($self, $handle, $state) = @_;
+
+	# texlive_base has junk which we need to clear
 	if ($handle->pkgname =~ m/^texlive_base-2009/) {
 		# we need to alter its packing-list
 		my $plist = OpenBSD::PackingList->from_installation(
@@ -240,6 +242,46 @@
 		    }, 
 		    '/usr/local/share/texmf-var');
 		quick_add($plist, '/usr/local/share/texmf/web2c/fmtutil.cnf');
+		quick_add($plist, '/usr/local/share/texmf/web2c/updmap.cfg');
+
+		# add links that were outside the plist in tl2009
+		my @links = qw(
+			lamed dvilualatex dviluatex lualatex metafun mfplain
+			amstex cslatex csplain eplain etex jadetex latex mex
+			mllatex mltex pdfcslatex pdfcsplain pdfetex pdfjadetex
+			pdflatex pdfmex pdfxmltex physe phyzzx texsis utf8mex
+			xmltex platex xelatex
+		);
+		foreach (@links) {
+			quick_add($plist, '/usr/local/bin/'.$_);
+		}
+
+		$plist->to_installation;
+	}
+
+	# nullify some scripts which can't run upgrading 2009 -> 2010
+	if ($handle->pkgname =~ m/^texlive_.*2009/) {
+		# we need to alter its packing-list
+		my $plist = OpenBSD::PackingList->from_installation(
+		    $handle->pkgname);
+
+		for my $i (@{$plist->{items}}) {
+			if ($i->isa("OpenBSD::PackingElement::Unexec")) {
+#					(($i->{name} =~ /tl-link-config$/) ||
+#					($i->{name} =~ /texlinks -u$/) ||
+#					($i->{name} =~ /texlinks$/))) {
+				print "nullifying: ".$i->{name}."\n";
+				bless $i, "OpenBSD::PackingElement::Comment";
+				$i->{name} = "quirk: ".$i->{name};
+			}
+			if (($i->isa("OpenBSD::PackingElement::Sample")) &&
+					(($i->{name} =~ /share\/texmf\/web2c\/updmap.cfg$/))) {
+				print "Found sample updmap: $i->{name}\n";
+				bless $i, "OpenBSD::PackingElement::Comment";
+				$i->{name} = "quirk: ".$i->{name};
+			}
+		}
+
 		$plist->to_installation;
 	}
 
