Index: README
===================================================================
RCS file: /cvs/ports/print/texlive/README,v
retrieving revision 1.1
diff -u -p -u -r1.1 README
--- README	8 Mar 2011 00:09:42 -0000	1.1
+++ README	22 May 2011 21:23:02 -0000
@@ -22,6 +22,7 @@ TeX Live 2010 - Porting progress.
    -> fix $HOME ls-R owned root bug:	done
    -> make fmtutil.cnf configs:		not needed this year
  * Implement remaining @xxx directives:	done
+ * emulate texlinks			done
  * Test lots:				done (several months)
  * Link to build as print/texlive:	done
 
Index: base/Makefile
===================================================================
RCS file: /cvs/ports/print/texlive/base/Makefile,v
retrieving revision 1.47
diff -u -p -u -r1.47 Makefile
--- base/Makefile	11 May 2011 07:45:53 -0000	1.47
+++ base/Makefile	22 May 2011 21:23:02 -0000
@@ -4,7 +4,7 @@ COMMENT =		base binaries for TeXLive typ
 
 DISTNAME = 		texlive-${DIST_V}-source
 PKGNAME =		texlive_base-${V}
-REVISION =		1
+REVISION =		2
 
 SHARED_LIBS =		kpathsea	3.0
 SHARED_LIBS +=		ptexenc		0.0
@@ -106,6 +106,19 @@ SUBST_VARS +=		TRUEPREFIX LOCALBASE
 pre-install:
 	mkdir ${WRKSRC}/inst
 
+# symlinks that would have been installed by texlinks.
+# texlinks can not do this itself in the fake dir, as it requires
+# a fully hashed texmf tree and working kpse configuration.
+ALEPH_LN =	lamed 
+LUATEX_LN =	dvilualatex dviluatex lualatex
+MPOST_LN =	metafun mfplain
+PDFTEX_LN =	amstex cslatex csplain eplain etex jadetex latex mex \
+		mllatex mltex pdfcslatex pdfcsplain pdfetex pdfjadetex \
+		pdflatex pdfmex pdfxmltex physe phyzzx texsis utf8mex \
+		xmltex
+PTEX_LN =	platex
+XETEX_LN =	xelatex
+
 post-install:
 	rm -Rf ${WRKSRC}/inst
 
@@ -121,4 +134,25 @@ post-install:
 		fi \
 	done
 
+	# emulate texlinks
+	for i in ${ALEPH_LN}; do \
+		ln -s ${TRUEPREFIX}/bin/aleph ${PREFIX}/bin/$$i; \
+	done
+	for i in ${LUATEX_LN}; do \
+		ln -s ${TRUEPREFIX}/bin/luatex ${PREFIX}/bin/$$i; \
+	done
+	for i in ${MPOST_LN}; do \
+		ln -s ${TRUEPREFIX}/bin/mpost ${PREFIX}/bin/$$i; \
+	done
+	for i in ${PDFTEX_LN}; do \
+		ln -s ${TRUEPREFIX}/bin/pdftex ${PREFIX}/bin/$$i; \
+	done
+	for i in ${PTEX_LN}; do \
+		ln -s ${TRUEPREFIX}/bin/ptex ${PREFIX}/bin/$$i; \
+	done
+	for i in ${XETEX_LN}; do \
+		ln -s ${TRUEPREFIX}/bin/xetex ${PREFIX}/bin/$$i; \
+	done
+
 .include <bsd.port.mk>
+
Index: base/pkg/PLIST
===================================================================
RCS file: /cvs/ports/print/texlive/base/pkg/PLIST,v
retrieving revision 1.17
diff -u -p -u -r1.17 PLIST
--- base/pkg/PLIST	11 May 2011 07:45:53 -0000	1.17
+++ base/pkg/PLIST	22 May 2011 21:23:02 -0000
@@ -3,9 +3,6 @@
 @conflict ht-<2
 @pkgpath print/teTeX/base
 @pkgpath print/teTeX/base,-fmt
-@comment mptopdf must be before texlinks -u
-bin/mptopdf
-@unexec texlinks -u
 bin/a2ping
 bin/afm2afm
 @bin bin/afm2pl
@@ -14,6 +11,7 @@ bin/afm2afm
 bin/allcm
 bin/allec
 bin/allneeded
+bin/amstex
 bin/arlatex
 bin/authorindex
 bin/autoinst
@@ -42,6 +40,8 @@ bin/cefspdflatex
 bin/chkweb
 bin/cmap2enc
 bin/context
+bin/cslatex
+bin/csplain
 @bin bin/ctangle
 @bin bin/ctie
 bin/ctxtools
@@ -64,6 +64,8 @@ bin/dvihp
 @bin bin/dvilj4
 @bin bin/dvilj4l
 bin/dvilj6
+bin/dvilualatex
+bin/dviluatex
 bin/dvipdfm
 @bin bin/dvipdfmx
 bin/dvipdft
@@ -79,9 +81,11 @@ bin/dvitomp
 bin/e2pall
 bin/ebb
 bin/ebong
+bin/eplain
 bin/epspdf
 bin/epspdftk
 bin/epstopdf
+bin/etex
 @bin bin/extconv
 bin/extractbb
 bin/fig4latex
@@ -106,6 +110,7 @@ bin/httex
 bin/httexi
 bin/htxelatex
 bin/htxetex
+bin/jadetex
 @bin bin/kpseaccess
 bin/kpsepath
 @bin bin/kpsereadlink
@@ -115,12 +120,15 @@ bin/kpsewhere
 @bin bin/kpsewhich
 bin/kpsexpand
 @bin bin/lacheck
+bin/lamed
+bin/latex
 bin/latex2man
 bin/latexdiff
 bin/latexdiff-vc
 @comment bin/latexmk
 bin/latexrevise
 bin/listings-ext.sh
+bin/lualatex
 @bin bin/luatex
 bin/luatools
 @bin bin/mag
@@ -129,8 +137,11 @@ bin/makeglossaries
 @bin bin/makejvf
 bin/mathspic
 @bin bin/mendex
+bin/metafun
+bin/mex
 @bin bin/mf
 @bin bin/mf-nowin
+bin/mfplain
 @bin bin/mft
 bin/mk4ht
 bin/mkgrkindex
@@ -145,11 +156,14 @@ bin/mktexlsr
 bin/mktexmf
 bin/mktexpk
 bin/mktextfm
+bin/mllatex
+bin/mltex
 @bin bin/mmafm
 @bin bin/mmpfb
 @bin bin/mpost
 %%SHARED%%
 %%xindy%%
+bin/mptopdf
 bin/mtxrun
 @bin bin/musixflx
 @bin bin/odvicopy
@@ -175,27 +189,37 @@ bin/pdfatfi
 @comment bin/pdfbook
 @bin bin/pdfclose
 bin/pdfcrop
+bin/pdfcslatex
+bin/pdfcsplain
+bin/pdfetex
 @comment bin/pdfflip
+bin/pdfjadetex
 @comment bin/pdfjam
 @comment bin/pdfjam-pocketmod
 @comment bin/pdfjam-slides3up
 @comment bin/pdfjam-slides6up
 @comment bin/pdfjoin
+bin/pdflatex
+bin/pdfmex
 @comment bin/pdfnup
 @bin bin/pdfopen
 @comment bin/pdfpun
 @bin bin/pdftex
 bin/pdfthumb
 @bin bin/pdftosrc
+bin/pdfxmltex
 @bin bin/pdvitype
 bin/perltex
 @bin bin/pfb2pfa
 bin/pfm2kpx
+bin/physe
+bin/phyzzx
 @bin bin/pk2bm
 bin/pkfix
 bin/pkfix-helper
 @bin bin/pktogf
 @bin bin/pktype
+bin/platex
 @bin bin/pltotf
 @bin bin/pooltype
 @bin bin/ppltotf
@@ -249,11 +273,10 @@ bin/texloganalyser
 bin/texlua
 bin/texluac
 bin/texmfstart
+bin/texsis
 @bin bin/tftopl
 bin/thumbpdf
 @bin bin/tie
-@comment bin/tl-link-config
-@comment bin/tl-link-config.beforesubst
 @comment bin/tlmgr
 bin/tpic2pdftex
 @bin bin/ttf2afm
@@ -264,6 +287,7 @@ bin/tpic2pdftex
 bin/ulqda
 bin/updmap
 bin/updmap-sys
+bin/utf8mex
 @bin bin/vftovp
 @bin bin/vlna
 bin/vpe
@@ -274,7 +298,9 @@ bin/vpl2vpl
 bin/xdvi
 @bin bin/xdvi-xaw
 @bin bin/xdvipdfmx
+bin/xelatex
 @bin bin/xetex
+bin/xmltex
 include/kpathsea/
 include/kpathsea/absolute.h
 include/kpathsea/c-auto.h
@@ -489,7 +515,6 @@ lib/libptexenc.la
 share/doc/pkg-readmes/${FULLPKGNAME}
 share/lcdf-typetools-for-tex-live/
 @exec mktexlsr > /dev/null 2>&1
-@exec texlinks
 @extraunexec rm -Rf %D/share/texmf-var/
 @extraunexec rm -Rf %D/share/texmf-config/
 @comment share/texmf-dist/bibtex/csf/base/88591lat.csf
Index: texmf/Makefile
===================================================================
RCS file: /cvs/ports/print/texlive/texmf/Makefile,v
retrieving revision 1.18
diff -u -p -u -r1.18 Makefile
--- texmf/Makefile	9 Mar 2011 14:43:59 -0000	1.18
+++ texmf/Makefile	22 May 2011 21:23:02 -0000
@@ -7,13 +7,14 @@ COMMENT-docs =		texlive documentation
 DISTNAME =		texlive-${DIST_V}-texmf
 MASTER_SITES0 =		http://theunixzoo.co.uk/distfiles/
 DISTFILES =		${DISTNAME}${EXTRACT_SUFX} \
-			texmf-var-pregen-2010${EXTRACT_SUFX}:0
+			texmf-var-pregen-2010.1${EXTRACT_SUFX}:0
 
 PKGNAME-main = 		texlive_texmf-minimal-${V}
 PKGNAME-full = 		texlive_texmf-full-${V}
 PKGNAME-docs =	 	texlive_texmf-docs-${V}
 
-REVISION-main =		0
+REVISION-main =		1
+REVISION-full =		0
 
 NO_BUILD =		Yes
 NO_REGRESS =		Yes
Index: texmf/distinfo
===================================================================
RCS file: /cvs/ports/print/texlive/texmf/distinfo,v
retrieving revision 1.3
diff -u -p -u -r1.3 distinfo
--- texmf/distinfo	8 Mar 2011 00:09:42 -0000	1.3
+++ texmf/distinfo	22 May 2011 21:23:02 -0000
@@ -1,10 +1,10 @@
 MD5 (texlive-20100722-texmf.tar.xz) = Et7DoyPE63W5ijFgxi3IZQ==
-MD5 (texmf-var-pregen-2010.tar.xz) = qNdvyLW7EldON2CP9GxnGQ==
+MD5 (texmf-var-pregen-2010.1.tar.xz) = VrbNMlORaiy5XzdYvPQG/g==
 RMD160 (texlive-20100722-texmf.tar.xz) = YIwA8D1MDvCOlX7ABxIybnXu4E4=
-RMD160 (texmf-var-pregen-2010.tar.xz) = i1n0mVovkMvxowFyk0QXro2b+XE=
+RMD160 (texmf-var-pregen-2010.1.tar.xz) = jdgnDV7xK0YCiyuFPn3eN+3tDVQ=
 SHA1 (texlive-20100722-texmf.tar.xz) = 32ugUJcFkvD184y53TbU0NZx+oA=
-SHA1 (texmf-var-pregen-2010.tar.xz) = pjLgp19E0fzxB0NQM64Vq3YQMZw=
+SHA1 (texmf-var-pregen-2010.1.tar.xz) = xzw3UZ69w105P/Svq05i20nWVtM=
 SHA256 (texlive-20100722-texmf.tar.xz) = LTED3JEAqevQDbtQDWeInaLj096J+N8RGA1bSlLgD4U=
-SHA256 (texmf-var-pregen-2010.tar.xz) = na3foQLh9WAqHvo21+vXu6l8JA+MafsszEonH9zqZWY=
+SHA256 (texmf-var-pregen-2010.1.tar.xz) = 6fDlUuDwbkPTJEntM1REVDs/xNqtrWhjy8p6g+jsLlg=
 SIZE (texlive-20100722-texmf.tar.xz) = 1137115288
-SIZE (texmf-var-pregen-2010.tar.xz) = 13799604
+SIZE (texmf-var-pregen-2010.1.tar.xz) = 13086664
Index: texmf/pkg/PLIST-full
===================================================================
RCS file: /cvs/ports/print/texlive/texmf/pkg/PLIST-full,v
retrieving revision 1.11
diff -u -p -u -r1.11 PLIST-full
--- texmf/pkg/PLIST-full	8 Mar 2011 00:09:42 -0000	1.11
+++ texmf/pkg/PLIST-full	22 May 2011 21:23:03 -0000
@@ -28114,6 +28114,4 @@ share/texmf/xindy/modules/tex/inputenc/l
 share/texmf/xindy/modules/tex/inputenc/latin2.xdy
 share/texmf/xindy/modules/tex/inputenc/latin3.xdy
 @exec mktexlsr > /dev/null 2>&1
-@exec texlinks
 @unexec-delete mktexlsr > /dev/null 2>&1
-@unexec-delete texlinks
Index: texmf/pkg/PLIST-main
===================================================================
RCS file: /cvs/ports/print/texlive/texmf/pkg/PLIST-main,v
retrieving revision 1.8
diff -u -p -u -r1.8 PLIST-main
--- texmf/pkg/PLIST-main	8 Mar 2011 00:09:42 -0000	1.8
+++ texmf/pkg/PLIST-main	22 May 2011 21:23:03 -0000
@@ -35003,7 +35003,7 @@ share/texmf/web2c/tcvn-t5.tcx
 share/texmf/web2c/texmf.cnf
 share/texmf/web2c/texmfcnf.lua
 share/texmf/web2c/updmap-hdr.cfg
-@sample share/texmf/web2c/updmap.cfg
+share/texmf/web2c/updmap.cfg
 share/texmf/web2c/viscii-t5.tcx
 share/texmf/xdvi/
 share/texmf/xdvi/XDvi
@@ -35013,13 +35013,10 @@ share/texmf/xdvi/pixmap/toolbar2.xpm
 share/texmf/xdvi/xdvi.cfg
 share/texmf-var/
 share/texmf-var/web2c/
-share/texmf-var/web2c/luatex/
-share/texmf-var/web2c/luatex/dviluatex.fmt
-share/texmf-var/web2c/luatex/dvilualatex.fmt
-share/texmf-var/web2c/luatex/lualatex.fmt
-share/texmf-var/web2c/luatex/luatex.fmt
+share/texmf-var/web2c/aleph/
+share/texmf-var/web2c/aleph/aleph.fmt
+share/texmf-var/web2c/aleph/lamed.fmt
 share/texmf-var/web2c/pdftex/
-share/texmf-var/web2c/pdftex/pdfetex.fmt
 share/texmf-var/web2c/pdftex/amstex.fmt
 share/texmf-var/web2c/pdftex/cont-en.fmt
 share/texmf-var/web2c/pdftex/cslatex.fmt
@@ -35027,6 +35024,8 @@ share/texmf-var/web2c/pdftex/pdfcslatex.
 share/texmf-var/web2c/pdftex/csplain.fmt
 share/texmf-var/web2c/pdftex/pdfcsplain.fmt
 share/texmf-var/web2c/pdftex/eplain.fmt
+share/texmf-var/web2c/pdftex/jadetex.fmt
+share/texmf-var/web2c/pdftex/pdfjadetex.fmt
 share/texmf-var/web2c/pdftex/latex.fmt
 share/texmf-var/web2c/pdftex/pdflatex.fmt
 share/texmf-var/web2c/pdftex/mex.fmt
@@ -35037,18 +35036,26 @@ share/texmf-var/web2c/pdftex/mllatex.fmt
 share/texmf-var/web2c/pdftex/mptopdf.fmt
 share/texmf-var/web2c/pdftex/pdftex.fmt
 share/texmf-var/web2c/pdftex/etex.fmt
+share/texmf-var/web2c/pdftex/pdfetex.fmt
+share/texmf-var/web2c/pdftex/physe.fmt
+share/texmf-var/web2c/pdftex/phyzzx.fmt
+share/texmf-var/web2c/pdftex/texsis.fmt
+share/texmf-var/web2c/pdftex/xmltex.fmt
+share/texmf-var/web2c/pdftex/pdfxmltex.fmt
 share/texmf-var/web2c/xetex/
 share/texmf-var/web2c/xetex/cont-en.fmt
 share/texmf-var/web2c/xetex/xetex.fmt
 share/texmf-var/web2c/xetex/xelatex.fmt
 share/texmf-var/web2c/metapost/
-share/texmf-var/web2c/metapost/mpost.mem
 share/texmf-var/web2c/metapost/metafun.mem
+share/texmf-var/web2c/metapost/mpost.mem
 share/texmf-var/web2c/metapost/mplib-luatex.mem
 share/texmf-var/web2c/metapost/mfplain.mem
-share/texmf-var/web2c/aleph/
-share/texmf-var/web2c/aleph/aleph.fmt
-share/texmf-var/web2c/aleph/lamed.fmt
+share/texmf-var/web2c/luatex/
+share/texmf-var/web2c/luatex/dvilualatex.fmt
+share/texmf-var/web2c/luatex/lualatex.fmt
+share/texmf-var/web2c/luatex/luatex.fmt
+share/texmf-var/web2c/luatex/dviluatex.fmt
 share/texmf-var/web2c/metafont/
 share/texmf-var/web2c/metafont/mf.base
 share/texmf-var/web2c/ptex/
@@ -35076,3 +35083,4 @@ share/texmf-var/fonts/map/dvipdfm/updmap
 share/texmf-var/fonts/map/dvipdfm/updmap/dvipdfm_dl14.map
 share/texmf-var/fonts/map/dvipdfm/updmap/dvipdfm_ndl14.map
 share/texmf-var/fonts/map/dvipdfm/updmap/dvipdfm.map
+@extra share/texmf-var/ls-R 
